src/ray/common/test_util.cc

```diff
@@ -41,7 +41,9 @@ void RedisServiceManagerForTest::TearDownTestCase() {
   std::string stop_redis_command =
       REDIS_CLIENT_EXEC_PATH + " -p " + std::to_string(REDIS_SERVER_PORT) + " shutdown";
   RAY_LOG(INFO) << "Stop redis command is: " << stop_redis_command;
-  RAY_CHECK(system(stop_redis_command.c_str()) == 0);
+  if (system(stop_redis_command.c_str()) != 0) {
+    RAY_LOG(WARNING) << "Failed to stop redis. The redis process may no longer exist.";
```

Comment:
why would this happen? is it because the shutdown sequence is not-deterministic, or it's possible that redis crashed?

Reply:
In order to test the redis exception scenario, TearDownTestCase will be called actively in TestGcsRedisFailureDetector testcase. And when TestGcsRedisFailureDetector testcase finished, it will call TearDownTestCase again but redis process is not exist.